src/components/TradeAssetSearch/hooks/useGetCustomTokensQuery.tsx

```diff
@@ -0,0 +1,62 @@
+import { type ChainId } from '@shapeshiftoss/caip'
+import { useQueries, type UseQueryResult } from '@tanstack/react-query'
+import type { TokenMetadataResponse } from 'alchemy-sdk'
+import { useCallback } from 'react'
+import { mergeQueryOutputs } from 'react-queries/helpers'
+import { isAddress } from 'viem'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { getAlchemyInstanceByChainId } from 'lib/alchemySdkInstance'
+
+type TokenMetadata = TokenMetadataResponse & {
+  chainId: ChainId
+  contractAddress: string
+  price: string
+}
+
+type UseGetCustomTokensQueryProps = {
+  contractAddress: string
+  chainIds: ChainId[]
+}
+
+type CustomTokenQueryKey = ['customTokens', string, ChainId]
+
+const getCustomTokenQueryKey = (contractAddress: string, chainId: ChainId): CustomTokenQueryKey => [
+  'customTokens',
+  contractAddress,
+  chainId,
+]
+
+export const useGetCustomTokensQuery = ({
+  contractAddress,
+  chainIds,
```

Comment:
blocking: don't use `chainIds` as a `queryKey`, but *only* the address. Using programmatic `chainIds` means:

- this will run on selected filter chain change
- related to the above, we will run this on non-EVM change selection

proposed solution: 
- do not pass `chainIds` altogether and use `evmChainIds` from `EvmBaseAdapter` instead. This way, the ChainIds list is static no matter the current chain selection
- additionally, `selectedChainId` (`ChainId` | 'All') could be passed down not to fetch altogether in case the selected chain is a non-EVM chain, ending up in a loading state that will end up with guaranteed empty results eventually

Reply:
I don't think this is quite right - `chainIds` is not the query key. Each query in `useQueries` has it's on query key, which includes the `chainId` and the `contractAddress`. This gives us the benefit of only running those queries that are necessary - e.g. if Polygon is selected, only the Polygon Alchemy request will be made. If "All" is then selected, we already have the Polygon request cached, and the rest will be fetched.

I've updated the logic a little in https://github.com/shapeshift/web/pull/7285/commits/3fd5e823b50d5d0f9575406abbed447affe36b7c to ensure only EVM chains are passed in.